### PR TITLE
avm2: Implement XML.setName

### PR DESCRIPTION
**Pull request by Lord-McSweeney**
Allows sprite costumes to be edited in Scratch 2 (progresses #12613).

### CODE REVIEW

**Comment by Dinnerbone** [comment]
Just to be clear; is this intended as a stub or full implementation? I'm not sure it handles qnames or processing instructions?

**Comment by Lord-McSweeney** [comment]
I didn't realize you could `setName` with a QName as the argument- I'll implement that now.

**Comment by Dinnerbone** [comment]
The implementation should follow 13.4.4.35 from this PDF: https://www.ecma-international.org/wp-content/uploads/ECMA-357_2nd_edition_december_2005.pdf (or search for `setName`)

**Comment by Lord-McSweeney** [comment]
I added support for QNames (but no namespaces).

**Review by sleepycatcoding — APPROVED** [review]
Looks good to me

File: tests/tests/swfs/avm2/xml_set_name/test.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+num_frames = 1
```
Comment (sleepycatcoding) [inline tests/tests/swfs/avm2/xml_set_name/test.toml:1]: A small nit, but I think according to the README.md in tests folder, the use of `num_frames` was deprecated and `num_ticks` is meant to be instead?

File: core/src/avm2/globals/xml.rs [hunk 2]
```diff
@@ -81,6 +81,42 @@ pub fn name<'gc>(
...
 
+pub fn set_name<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let xml = this.as_xml_object().unwrap();
+    let node = xml.node();
+
+    let new_name = args.get_value(0);
```
Comment (adrian17) [inline core/src/avm2/globals/xml.rs:92]: now that we have `isXMLName`, we should use it here and throw an exception if the name is illegal, right? (we can do this in a follow-up if you prefer)

**Review by adrian17 — APPROVED** [review]

File: core/src/avm2/globals/xml.rs (hunk 2 above)
Comment (Lord-McSweeney) [reply core/src/avm2/globals/xml.rs:92]: I would prefer to do it in this PR.

**Pull request merged**